Task Liverpool/src/main/java/pages/SearchResultsPage.java

```diff
@@ -0,0 +1,152 @@
+    //private By filterLowerThan$500Button2 = By.xpath("//input[@id='variants.prices.sortPrice-100-500']"); //Please ignore these locators, I am leaving them here for future reference
+    //private By filterLowerThan$500Button3 = By.cssSelector("#variants\\.prices\\.sortPrice-100-500"); //Please ignore these locators, I am leaving them here for future reference
+    private By noResultsText = By.cssSelector(".a-headline__noResults");
+    private By firstResult;
+    List<WebElement> pricesList;
+    List<Integer> pricesIntList;
+
+    public enum SortType {
+        RELEVANCE, LOWPRICE, HIGHPRICE, RATING, VIEWED, SOLD
+    }
+
+    public SearchResultsPage(WebDriver webDriver) {
+        super(webDriver);
+    }
+
+    public int getNumberOfResultsFromUpperRightLabel(){
+        return Integer.parseInt(getElementText(numberOfResultsFromUpperRightLabel));
+    }
+
+    //Logic for clicking on the Results Page Right Arrow
+    public boolean clickNextPage(){
+        if (isElementDisplayed(rightArrow)){
+            String productId = getElementAttribute(findElement(searchResultsDisplayed), "data-prodid");
+            firstResult = By.cssSelector("li.m-product__card[data-prodid=\"" + productId + "\"]");
+            scrollToElement(rightArrow);
+            click(rightArrow);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    public int getTotalNumberOfResultsDisplayed() {
+        int totalResults = findElements(searchResultsDisplayed).size();
+
+        while (clickNextPage()){
+            totalResults += findElementsAfterACurrentElementDisappear(searchResultsDisplayed, firstResult).size();
+        }
+        return totalResults;
+    }
+
+    public void clickListViewButton(){
+        click(listViewButton);
+    }
+
+    public void clickSortProductsBy (SortType sortType) throws InterruptedException {
+
+        click(sortByDropDownList);
+        switch (sortType) {
+            case RELEVANCE:
+                click(sortByRelevance);
+                break;
+
+            case LOWPRICE:
+                click(sortByLowPrice);
+                break;
+
+            case HIGHPRICE:
+                click(sortByHighPrice);
+                break;
+
+            case RATING:
+                click(sortByRating);
+                break;
+
+            case VIEWED:
+                click(sortByViewed);
+                break;
+
+            case SOLD:
+                click(sortBySold);
+                break;
+        }
+        Thread.sleep(1000);
+    }
+
+    public void getPricesInt(){
+        pricesList = findElements(productPrice);
+        pricesIntList = new ArrayList<>();
+
+        for (WebElement prices: pricesList) {
+            pricesIntList.add(Integer.parseInt(prices.getText().substring(1).replace(",", "")));
+        }
+    }
+
+    public void addValuesToPricesIntListFromMoreResultsPages(){
```

Comment:
maybe return something instead of adding to attribute... I think public here is not needed

Reply:
I am leaving it public because I am going to be calling this method from HomeTest (also because I moved the logic to compare and validate there). I can't see how this could work by returning something, if you can tell me I would appreciate it 👍 .